controllers/drcluster_controller.go

```diff
@@ -260,12 +262,83 @@ func (u *drclusterUpdater) clusterFenceHandle() error {
 	}
 
 	if u.object.Spec.ClusterFence == ramen.ClusterFenceStateFenced {
-		return fmt.Errorf("currently DRPolicy cant handle ClusterFenceStateFenced")
+		// return fmt.Errorf("currently DRPolicy cant handle ClusterFenceStateFenced")
+		return u.clusterFence()
 	}
 
 	return nil
 }
 
+func (u *drclusterUpdater) clusterFence() error {
```

Comment:
drClusterUpdater is only to update status, do not use it for overall DRCluster operations. IOW, keep these functions out of the updater, and only update status using the interface.

Reply:
ACK. Will do as part of a subsequent PR.